packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -141,29 +212,46 @@ export class QueuedRequestController extends BaseController<
    * requests, ensuring they are executed one after the other to prevent concurrency issues and maintain proper
    * execution flow.
    *
-   * @param requestNext - A function representing the request to be enqueued. It returns a promise that
+   * @param request - The JSON-RPC request to process.
+   * @param requestNext - A function representing the next steps for processing this request. It returns a promise that
    * resolves when the request is complete.
    * @returns A promise that resolves when the enqueued request and any subsequent asynchronous
    * operations are fully processed. This allows you to await the completion of the enqueued request before continuing
    * with additional actions. If there are multiple enqueued requests, this function ensures they are processed in
    * the order they were enqueued, guaranteeing sequential execution.
    */
-  async enqueueRequest(requestNext: (...arg: unknown[]) => Promise<unknown>) {
+  async enqueueRequest(
+    request: QueuedRequestMiddlewareJsonRpcRequest,
+    requestNext: (...arg: unknown[]) => Promise<unknown>,
+  ) {
     this.#updateCount(1);
-
     if (this.state.queuedRequestCount > 1) {
-      await this.currentRequest;
+      try {
+        await this.currentRequest;
+      } catch (_error) {
+        // error ignored - this is handled in the middleware instead
+        this.#updateCount(-1);
+      }
     }
 
-    this.currentRequest = requestNext()
-      .then(() => {
-        this.#updateCount(-1);
-      })
-      .catch((e) => {
+    const processCurrentRequest = async () => {
+      try {
+        if (
+          request.method !== 'wallet_switchEthereumChain' &&
+          request.method !== 'wallet_addEthereumChain'
+        ) {
+          await this.#switchNetworkIfNecessary(request);
+        }
+
+        await requestNext();
+      } finally {
+        // The count is updated as part of the request processing to ensure
```

Comment:
This comment seems out of date. Since `requestNext` is called above, doesn't that mean the count is decremented after all remaining requests in the middleware chain are run?

Reply:
Yes, it does mean that, but that doesn't contradict the comment. By "request processing" I'm referring to the name of this function: `processCurrentRequest`. The count gets updated as part of this function. This function is what the next request waits upon (see `this.currentRequest = processCurrentRequest();`)